app/services/foreman/deprecation.rb

```diff
@@ -2,12 +2,28 @@ module Foreman
   class Deprecation
     #deadline_version - is the version the deprecation is going to be deleted, the format must be a major release e.g "1.8"
     def self.deprecation_warning(foreman_version_deadline, info)
-      raise Foreman::Exception.new(N_("Invalid version format, please enter in x.y (only major version).")) unless foreman_version_deadline.to_s.match(/\A\d[.]\d+\z/)
+      check_version_format foreman_version_deadline
       ActiveSupport::Deprecation.warn("You are using a deprecated behavior, it will be removed in version #{foreman_version_deadline}, #{info}", caller(2))
     end
 
+    def self.check_version_format(foreman_version_deadline)
+      raise Foreman::Exception.new(N_("Invalid version format, please enter in x.y (only major version).")) unless foreman_version_deadline.to_s.match(/\A\d[.]\d+\z/)
+    end
+
     def self.api_deprecation_warning(info)
       ActiveSupport::Deprecation.warn("Your API call uses deprecated behavior, #{info}", caller)
     end
+
+    def self.renderer_deprecation(foreman_version_deadline, method, new_method)
+      check_version_format foreman_version_deadline
+      called_from_params = false
+      caller.each_with_index do |item, index|
+        called_from_params = true if item.match(/host_params\.rb.*params/)
+        return if called_from_params && item.match(/managed\.rb.*info/)
+        next unless item.match(/renderer\.rb.*render_safe/)
```

Comment:
When using `host_param` in a template, two deprecation warnings seem to be logged. Please check this works properly. A simple unit test that fails:

  test 'should render host param using "host_param" helper without deprecation' do
    @renderer.host = FactoryGirl.create(:host, :with_puppet)
    Rails.logger.expects(:warn).never
    assert @renderer.render_safe("<%= host_param('test') %>", DummyRenderer::ALLOWED_HELPERS).present?
  end

The logic on this line seems incorrect to me, it skips the deprecation warning _only_ from render_safe. Surely any other entry in the callstack would trigger this?

I'm unsure why template rendering triggers two deprecation warnings without further investigation, perhaps something else internally is calling `@host.params` and triggering a second. Please check it's working correctly in real usage.